plugins/extractors/metabase/metabase.go

```diff
@@ -147,125 +129,206 @@ func (e *Extractor) buildDashboard(d Dashboard) (data *assets.Dashboard, err err
-		ID string `json:"id"`
-	}
-	var data responseID
-	err = e.makeRequest("POST", e.config.Host+"/api/session", payload, &data)
-	if err != nil {
-		return
+		for _, upstream := range chart.Lineage.Upstreams {
+			if _, duplicate := existing[upstream.Urn]; duplicate {
+				continue
+			}
+
+			upstreams = append(upstreams, upstream)
+			existing[upstream.Urn] = true
+		}
 	}
 
-	return data.ID, nil
+	return
 }
 
-// helper function to avoid rewriting a request
-func (e *Extractor) makeRequest(method, url string, payload interface{}, data interface{}) (err error) {
-	jsonBytes, err := json.Marshal(payload)
-	if err != nil {
-		return errors.Wrap(err, "failed to encode the payload JSON")
+func (e *Extractor) extractDbComponent(database Database) (service, cluster, dbName string) {
+	service = database.Engine
+
+	switch service {
+	case "h2":
+		comps := strings.Split(database.Details.Db, ";")
+		dbUrlComps := strings.Split(comps[0], "/")
+
+		cluster = strings.Join(dbUrlComps[:len(dbUrlComps)-1], "/")
+		dbName = dbUrlComps[len(dbUrlComps)-1]
+	case "postgres":
+		fallthrough
+	case "mysql":
+		cluster = fmt.Sprintf("%s:%d", database.Details.Host, database.Details.Port)
+		dbName = database.Details.Dbname
+	case "bigquery":
+		cluster = database.Details.ProjectID
+		dbName = database.Details.DatasetID
+	default:
+		e.logger.Warn(fmt.Sprintf("unsupported database engine \"%s\"", service))
 	}
 
-	req, err := http.NewRequest(method, url, bytes.NewBuffer(jsonBytes))
-	if err != nil {
-		return errors.Wrap(err, "failed to create request")
-	}
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("X-Metabase-Session", e.sessionID)
+	return
+}
 
-	res, err := e.client.Do(req)
+func (e *Extractor) getTableNamesFromSQL(datasetQuery NativeDatasetQuery) (tableNames []string, err error) {
+	query, err := evaluateQueryTemplate(datasetQuery)
 	if err != nil {
-		return errors.Wrap(err, "failed to generate response")
-	}
-	if res.StatusCode >= 300 {
-		return fmt.Errorf("getting %d status code", res.StatusCode)
+		err = errors.Wrap(err, "error adding default value to template in query")
+		return
 	}
-
-	bytes, err := ioutil.ReadAll(res.Body)
+	tableNames, err = extractTableNamesFromSQL(query)
 	if err != nil {
-		return errors.Wrap(err, "failed to read response body")
-	}
-	if err = json.Unmarshal(bytes, &data); err != nil {
-		return errors.Wrapf(err, "failed to parse: %s", string(bytes))
+		err = errors.Wrap(err, "error when parsing SQL")
 	}
 
 	return
 }
 
+func (e *Extractor) buildURN(service, cluster, dbName, tableName string) string {
+	tableComps := strings.Split(tableName, ".")
+	compLength := len(tableComps)
+	tableName = tableComps[len(tableComps)-1]
+
+	switch service {
+	case "postgres":
+		if compLength > 1 && tableComps[0] != "public" {
+			cluster = tableComps[0]
+		}
+	case "bigquery":
+		if compLength > 2 {
+			cluster = tableComps[0]
+			dbName = tableComps[1]
+		} else if compLength > 1 {
+			dbName = tableComps[0]
+		}
+	}
```

Comment:
what would be the default case?

Reply:
@mabdh  the default case is there is no modification on `cluster` and `dbName` variables.
Do you think we should use `if else` instead of `switch` for better clarity?

Reply:
I see, no need to I guess. We can keep it like this.